File: Proyecto Migrar/MainApp/ClsFramework/ClsSistema.vb
Namespace SIS.Framework.Common

    Public Interface IAcciones

        Sub Consultar()
        Sub Adicionar()
        Sub Modificar()
        Sub Eliminar()
        Sub Imprimir()

    End Interface

    Public Class ClsSistema
        Private _Cusuario As String
        Private _NombreUsuario As String
        Private _ObjInfoData As DataQuery
        Private _ConexionBaseDato As String
        Private _NombreAplicacion As String
        Private _NombreAplicacionImpresion As String
        Private _CodigoOpcionSistema As String
        Private _NombreOpcionSistema As String
        Private _DescripcionOpcionSistema As String
        Private _RefrescaCamposenCarga As String
        Private _CodigoEmpresa As String
        Private _CodigoAnio As String
        Private _CodigoMes As String
        Private _NecesitaCodigoEmpresa As String
        Private _NecesitaPeriodoContable As String
        Private _ValidaPeriodoContable As String
        Private _ValidaPeriodoLogistico As String
        Private _CodigoEmpresaDefecto As String
        Private _EstadoUsuario As Object
        Private _TipoAutenticacionSistema As TipoAutenticacion
        Private _CodigoInstancia As String
        Private _NombreInstancia As String
        Private _CodigoBodegaRecepcion As String

        Private _ServidorDeReportes As String

        Private _ExcluirDocs As String

        Public Property CodigoBodegaRecepcion() As String
            Get
                Return _CodigoBodegaRecepcion
            End Get
            Set(ByVal value As String)
                _CodigoBodegaRecepcion = value
            End Set
        End Property

        Public Property NombreAplicacionImpresion() As String
            Get
                Return _NombreAplicacionImpresion
            End Get
            Set(ByVal value As String)
                _NombreAplicacionImpresion = value
            End Set
        End Property

        Public Property CodigoInstancia() As String
            Get
                Return _CodigoInstancia
            End Get
            Set(ByVal value As String)
                _CodigoInstancia = value
            End Set
        End Property

        Public Property NombreInstancia() As String
            Get
                Return _NombreInstancia
            End Get
            Set(ByVal value As String)
                _NombreInstancia = value
            End Set
        End Property

        Public Property TipoAutenticacionSistema() As TipoAutenticacion
            Get
                Return _TipoAutenticacionSistema
            End Get
            Set(ByVal value As TipoAutenticacion)
                _TipoAutenticacionSistema = value
            End Set
        End Property

        Public Property ValidaPeriodoContable() As String
            Get
                Return _ValidaPeriodoContable
            End Get
            Set(ByVal value As String)
                _ValidaPeriodoContable = value
            End Set
        End Property

        Public Property ValidaPeriodoLogistico() As String
            Get
                Return _ValidaPeriodoLogistico
            End Get
            Set(ByVal value As String)
                _ValidaPeriodoLogistico = value
            End Set
        End Property

        Public Property EstadoUsuario() As String
            Get
                Return _EstadoUsuario
            End Get
            Set(ByVal value As String)
                _EstadoUsuario = value
            End Set
        End Property

        Public Property CodigoEmpresaDefecto() As String
            Get
                Return _CodigoEmpresaDefecto
            End Get
            Set(ByVal value As String)
                _CodigoEmpresaDefecto = value
            End Set
        End Property

        Public Property DescripcionOpcionSistema() As String
            Get
                Return _DescripcionOpcionSistema
            End Get
            Set(ByVal value As String)
                _DescripcionOpcionSistema = value
            End Set
        End Property
        Public Property NecesitaCodigoEmpresa() As String
            Get
                Return _NecesitaCodigoEmpresa
            End Get
            Set(ByVal value As String)
                _NecesitaCodigoEmpresa = value
            End Set
        End Property

        Public Property NecesitaPeriodoContable() As String
            Get
                Return _NecesitaPeriodoContable
            End Get
            Set(ByVal value As String)
                _NecesitaPeriodoContable = value
            End Set
        End Property

        Public Property CodigoEmpresa() As String
            Get
                Return _CodigoEmpresa
            End Get
            Set(ByVal value As String)
                _CodigoEmpresa = value
            End Set
        End Property

        Public Property CodigoAnio() As String
            Get
                Return _CodigoAnio
            End Get
            Set(ByVal value As String)
                _CodigoAnio = value
            End Set
        End Property

        Public Property CodigoMes() As String
            Get
                Return _CodigoMes
            End Get
            Set(ByVal value As String)
                _CodigoMes = value
            End Set
        End Property

        Public Property RefrescaCamposenCarga() As String
            Get
                Return _RefrescaCamposenCarga
            End Get
            Set(ByVal value As String)
                _RefrescaCamposenCarga = value
            End Set
        End Property

        Public Property CodigoOpcionSistema() As String
            Get
                Return _CodigoOpcionSistema
            End Get
            Set(ByVal value As String)
                _CodigoOpcionSistema = value
            End Set
        End Property
        Public Property NombreOpcionSistema() As String
            Get
                Return _NombreOpcionSistema

            End Get
            Set(ByVal value As String)
                _NombreOpcionSistema = value
            End Set
        End Property
        Public Property NombreAplicacion() As String
            Get
                Return _NombreAplicacion
            End Get
            Set(ByVal value As String)
                _NombreAplicacion = value
            End Set
        End Property
        Public Property ConexionBaseDato() As String
            Get
                Return _ConexionBaseDato
            End Get
            Set(ByVal value As String)
                _ConexionBaseDato = value
            End Set
        End Property
        Public Property Cusuario() As String
            Get
                Return _Cusuario
            End Get
            Set(ByVal value As String)
                _Cusuario = value
            End Set
        End Property
        Public Property NombreUsuario() As String
            Get
                Return _NombreUsuario

            End Get
            Set(ByVal value As String)
                _NombreUsuario = value
            End Set
        End Property

        Public Property ServidorDeReportes() As String
            Get
                Return _ServidorDeReportes
            End Get
            Set(ByVal value As String)
                _ServidorDeReportes = value
            End Set
        End Property

        Public Property ExcluirDocs() As String
            Get
                Return _ExcluirDocs
            End Get
            Set(ByVal value As String)
                _ExcluirDocs = value
            End Set
        End Property

        Public ReadOnly Property ObjInfoData() As DataQuery
            Get
                Return _ObjInfoData
            End Get
        End Property

        Sub New()
            _ObjInfoData = New DataQuery
        End Sub

    End Class

    Public Class DataQuery
        Public CodigoError As Decimal
        Public MensajeError As String
        Public ObjData As New DataSet
        Public Resultado As Boolean
        Public CodigoAuxiliar As Object
        Public FilasAfectadas As Integer

        Sub New()
            CodigoError = 0
            FilasAfectadas = 0
            MensajeError = ""
            CodigoAuxiliar = ""
            Resultado = False
        End Sub

    End Class

    Public Enum TipoUsuario
        Consulta = 1
        Normal = 2
        Supervisor = 3
        SuperUsuario = 4
        Administrador = 5
    End Enum

    Public Enum TipoActualizacion
        Adicionar = 1
        Actualizar = 2
        Eliminar = 3
        No_Definida = 4
    End Enum

    Public Enum TipoEstadoUsuario
        Pendiente = 0
        Activo = 1
        Suspendido = 2
    End Enum

    Public Enum TipoAutenticacion
        TablaSistema = 1
        ActiveDirectory = 2
        SQLServer = 3
    End Enum
End Namespace

File: SCSA09022015/SICSA/ClsFramework/ClsSistema.cs
using System.Data;

namespace SIS.Framework.Common
{
    
    public interface IAcciones
    {
        void Consultar();
        void Adicionar();
        void Modificar();
        void Eliminar();
        void Imprimir();
        void Limpiar();
    }

    public class ClsSistema
    {
        private string _Cusuario;
        private string _NombreUsuario;
        private DataQuery _ObjInfoData;
        private string _ConexionBaseDato;
        private string _NombreAplicacion;
        private string _NombreAplicacionImpresion;
        private string _CodigoOpcionSistema;
        private string _NombreOpcionSistema;
        private string _DescripcionOpcionSistema;
        private string _RefrescaCamposenCarga;
        private string _CodigoEmpresa;
        private string _CodigoAnio;
        private string _CodigoMes;
        private string _NecesitaCodigoEmpresa;
        private string _NecesitaPeriodoContable;
        private string _ValidaPeriodoContable;
        private string _CodigoEmpresaDefecto;
        private string _EstadoUsuario;
        //private  _TipoAutenticacionSistema As TipoAutenticacion
        private string _CodigoInstancia;
        private string _NombreInstancia;

        private string _ServidorDeReportes;


        public string NombreAplicacionImpresion{
            get
            {
                return _NombreAplicacionImpresion;
            }
            set
            {
                _NombreAplicacionImpresion = value;
            }

        }

        public string CodigoInstancia
        {
            get
            {
                return _CodigoInstancia;
            }
            set
            {
                _CodigoInstancia = value;
            }
        }

        public string NombreInstancia
        {
            get
            {
                return _NombreInstancia;
            }
            set
            {
                _NombreInstancia = value;
            }
        }

        public string ValidaPeriodoContable
        {
            get
            {
                return _ValidaPeriodoContable;
            }
            set
            {
                _ValidaPeriodoContable = value;
            }
        }

        public string EstadoUsuario
        {
            get
            {
                return _EstadoUsuario;
            }
            set
            {
                _EstadoUsuario = value;
            }
        }

        public string CodigoEmpresaDefecto
        {
            get
            {
                return _CodigoEmpresaDefecto;
            }
            set
            {
                _CodigoEmpresaDefecto = value;
            }
        }

        public string DescripcionOpcionSistema
        {
            get
            {
                return _DescripcionOpcionSistema;
            }
            set
            {
                _DescripcionOpcionSistema = value;
            }
        }
        public string NecesitaCodigoEmpresa
        {
            get
            {
                return _NecesitaCodigoEmpresa;
            }
            set
            {
                _NecesitaCodigoEmpresa = value;
            }
        }

        public string NecesitaPeriodoContable
        {
            get
            {
                return _NecesitaPeriodoContable;
            }
            set
            {
                _NecesitaPeriodoContable = value;
            }
        }

        public string CodigoEmpresa
        {
            get
            {
                return _CodigoEmpresa;
            }
            set
            {
                _CodigoEmpresa = value;
            }
        }

        public string CodigoAnio
        {
            get
            {
                return _CodigoAnio;
            }
            set
            {
                _CodigoAnio = value;
            }
        }

        public string CodigoMes
        {
            get
            {
                return _CodigoMes;
            }
            set
            {
                _CodigoMes = value;
            }
        }

        public string RefrescaCamposenCarga
        {
            get
            {
                return _RefrescaCamposenCarga;
            }
            set
            {
                _RefrescaCamposenCarga = value;
            }
        }

        public string CodigoOpcionSistema
        {
            get
            {
                return _CodigoOpcionSistema;
            }
            set
            {
                _CodigoOpcionSistema = value;
            }
        }
        public string NombreOpcionSistema
        {
            get
            {
                return _NombreOpcionSistema;

            }
            set
            {
                _NombreOpcionSistema = value;
            }
        }
        public string NombreAplicacion
        {
            get
            {
                return _NombreAplicacion;
            }
            set
            {
                _NombreAplicacion = value;
            }
        }
        public string ConexionBaseDato
        {
            get
            {
                return _ConexionBaseDato;
            }
            set
            {
                _ConexionBaseDato = value;
            }
        }
        public string Cusuario
        {
            get
            {
                return _Cusuario;
            }
            set
            {
                _Cusuario = value;
            }
        }
        public string NombreUsuario
        {
            get
            {
                return _NombreUsuario;

            }
            set
            {
                _NombreUsuario = value;
            }
        }

        public string ServidorDeReportes
        {
            get
            {
                return _ServidorDeReportes;
            }
            set
            {
                _ServidorDeReportes = value;
            }
        }

        public DataQuery ObjInfoData {
            get{
                return _ObjInfoData;
            }
        }

        public ClsSistema()
        {
            _ObjInfoData = new DataQuery();
        }
    }

    public class DataQuery
    {
        public decimal CodigoError;
        public string MensajeError;
        public DataSet ObjData;
        public bool Resultado;
        public object CodigoAuxiliar;
        public int FilasAfectadas;

        public DataQuery()
        {
            CodigoError = 0;
            FilasAfectadas = 0;
            MensajeError = "";
            CodigoAuxiliar = "";
            Resultado = false;
        }
    }

    public enum TipoActualizacion
    {
        Adicionar = 1,
        Actualizar = 2,
        Eliminar = 3,
        No_Definida = 4
    }
}